tests/evaluate_sentence_splitter.py

```diff
@@ -0,0 +1,90 @@
+import re
+from collections.abc import Generator
+from pathlib import Path
+
+import spacy
+import typer
+from tqdm import tqdm
+
+from fun_sentence_splitter.sentence_splitter import Span, init
+
+WHITE_SPACE = re.compile(r"\s+")
+
+
+def main(
+        data_dir: Path,
+        max_len: int = 100,
+        spacy_model: str = "de_core_news_sm",
+) -> None:
+    sentence_splitter = init(
+        max_len_before_split=max_len,
+        spacy_model=spacy_model,
+    )
+
+    tp, fp, fn = 0, 0, 0
+    spans_count = 0
+    texts_count = 0
+    for split_file in tqdm(list(data_dir.glob("*.split"))):
+        text_file = split_file.with_suffix(".txt")
+        gold_spans = frozenset(_find_spans(text_file=text_file, split_file=split_file))
+        test_spans = frozenset(sentence.span for sentence in sentence_splitter(text_file.read_text()))
+        tp_, fp_, fn_ = _count_overlap(gold_spans, test_spans)
+        tp += tp_
+        fp += fp_
+        fn += fn_
+        spans_count += len(gold_spans)
+        texts_count += 1
+    weighted_avg_f1 = _f1(tp, fp, fn)
+
+    print(f"\n{80 * '-'}")
+    print(f"f1 using spacy {spacy_model}@{spacy.__version__}: {weighted_avg_f1:.5} ({spans_count} spans from {texts_count} files)")  # noqa: E501
+    print(f"{80 * '-'}\n")
+
+
+def _find_spans(text_file: Path, split_file: Path) -> Generator[Span, None, None]:
+    if not text_file.exists():
+        print(f"ERROR - File {text_file} not found")
+        return
+
+    if not split_file.exists():
+        print(f"ERROR - File {split_file} not found")
+        return
+
+    full_text = text_file.read_text()
+    stop_idx = 0
+    with split_file.open() as f:
+        for sentence in f.read().splitlines():
+            # replace all spaces with regex pattern
+            tmp = WHITE_SPACE.sub(r" ", sentence).strip()
+            tmp = re.escape(tmp)
+            tmp = tmp.replace(r"\ ", r"\s+")
+            re_sentence = re.compile(tmp)
+            match = re_sentence.search(full_text, stop_idx)
+            if not match:
+                print(f'ERROR - Could not "{re_sentence.pattern}" after {stop_idx} in {text_file}')
+                continue
+            start_idx, stop_idx = match.start(), match.end()
+            yield start_idx, stop_idx
+
+
+def _count_overlap(
+        gold_split: frozenset[Span],
+        test_split: frozenset[Span],
+) -> tuple[int, int, int]:
+    tp = len(gold_split | test_split)
```

Comment:
tp should be Schnittmenge, not union :grimacing: 
```suggestion
    tp = len(gold_split & test_split)
```